Sources/App/Commands/Analyze.swift

```diff
@@ -354,21 +364,17 @@ func diffVersions(client: Client,
                   logger: Logger,
                   threadPool: NIOThreadPool,
                   transaction: Database,
-                  package: Package) -> EventLoopFuture<(toAdd: [Version],
-                                                        toDelete: [Version],
-                                                        toKeep: [Version])> {
+                  package: Package) -> EventLoopFuture<VersionDelta> {
     guard let cacheDir = Current.fileManager.cacheDirectoryPath(for: package) else {
         return transaction.eventLoop.future(error: AppError.invalidPackageCachePath(package.id, package.url))
     }
     guard let pkgId = package.id else {
         return transaction.eventLoop.future(error: AppError.genericError(nil, "PANIC: package id nil for package \(package.url)"))
     }
-    
-    let defaultBranch = Repository.defaultBranch(on: transaction, for: package)
-        .map { b -> [Reference] in
-            if let b = b { return [.branch(b)] } else { return [] }  // drop nil default branch
-        }
```

Comment:
This was an unnecessary query we ran against the `repositories` table. We were already joining against it. I dropped the whole `Repository.defaultBranch` function.